go/rootprogram/issuers.go

```diff
@@ -46,18 +49,69 @@ type EnrolledIssuer struct {
 type MozIssuers struct {
 	issuerMap map[string]IssuerData
 	mutex     *sync.Mutex
+	diskPath  string
+	reportUrl string
 }
 
 func NewMozillaIssuers() *MozIssuers {
 	return &MozIssuers{
 		issuerMap: make(map[string]IssuerData, 0),
 		mutex:     &sync.Mutex{},
+		diskPath:  "/tmp/mozissuers",
+		reportUrl: kMozCCADBReport,
 	}
 }
 
+type verifier struct {
+}
+
+func (v *verifier) IsValid(path string) error {
+	mi := NewMozillaIssuers()
+	return mi.LoadFromDisk(path)
+}
+
+type loggingAuditor struct{}
+
+func (ta *loggingAuditor) FailedDownload(issuer downloader.DownloadIdentifier, crlUrl *url.URL, dlTracer *downloader.DownloadTracer, err error) {
+	glog.Warningf("Failed download of %s: %s", crlUrl.String(), err)
+}
+func (ta *loggingAuditor) FailedVerifyUrl(issuer downloader.DownloadIdentifier, crlUrl *url.URL, dlTracer *downloader.DownloadTracer, err error) {
+	glog.Warningf("Failed verify of %s: %s", crlUrl.String(), err)
+}
+func (ta *loggingAuditor) FailedVerifyPath(issuer downloader.DownloadIdentifier, crlPath string, err error) {
+	glog.Warningf("Failed verify of %s: %s", crlPath, err)
+}
+
+type identifier struct{}
+
+func (i *identifier) ID() string {
+	return "Mozilla Issuers"
+}
+
 func (mi *MozIssuers) Load() error {
-	// TODO: Use a local cache
-	return mi.downloadAndParse(kMozCCADBReport)
+	ctx := context.Background()
+
+	display := mpb.New(
+		mpb.WithOutput(ioutil.Discard),
+	)
+
+	dataUrl, err := url.Parse(mi.reportUrl)
+	if err != nil {
+		glog.Fatalf("Couldn't parse CCADB URL of %s: %s", mi.reportUrl, err)
```

Comment:
Do we want to `return err` here?

Reply:
I probably should, for clarity. Fatal will terminate as-is, it's possible `go vet` will complain with the unreachable code, but I'll handle it. 

A further question is whether it should Fatal, but IMO since this is a configuration error, it should indeed.

Reply:
For clarity, sure. `glog.Fatal` aborts the program, but `go vet` doesn't complain about unreachable code, so SGTM!

(As I mentioned in the other comment, I think this does deserve an abort for the URL being misconfigured.)

Reply:
I even looked at the documentation, but apparently I skimmed it too quickly and missed that it calls `os.Exit(255)`. In that case I don't mind either way, but having the return statement doesn't hurt.